src/pages/beneficiaries/Beneficiaries.js

```diff
@@ -12,71 +15,228 @@ const FILTER_MAP = {
-                setBeneficiary(data);
-                console.log(beneficiary);
+                setBeneficiaries(data);
+                console.log("beneficiaries: " + beneficiaries);
             } catch (error) {
                 console.error(error);
             }
         });
     };
 
-    const sortByName = () => {
-        let data = [...beneficiary];
+    const sortByFirstName = () => {
+        let data = [...beneficiaries];
         data.sort((a, b) => a.firstName.localeCompare(b.firstName));
-        setBeneficiary(data);
+        setBeneficiaries(data);
+    };
+
+    const sortBackwardsByFirstName = () => {
+        let data = [...beneficiaries];
+        data.sort((a, b) => b.firstName.localeCompare(a.firstName));
+        setBeneficiaries(data);
+    };
+
+    const sortByLastName = () => {
+        let data = [...beneficiaries];
+        data.sort((a, b) => a.lastName.localeCompare(b.lastName));
+        setBeneficiaries(data);
+    };
+
+    const sortBackwardsByLastName = () => {
+        let data = [...beneficiaries];
+        data.sort((a, b) => b.lastName.localeCompare(a.lastName));
+        setBeneficiaries(data);
     };
 
     const sortByDate = () => {
-        let data = [...beneficiary];
+        let data = [...beneficiaries];
         data.sort(function (a, b) {
             const dateA = new Date(a.joinDate);
             const dateB = new Date(b.joinDate);
             return dateB - dateA;
         });
-        console.log(data);
-        setBeneficiary(data);
+        setBeneficiaries(data);
+    };
+
+    const sortBackwardsByDate = () => {
+        let data = [...beneficiaries];
+        data.sort(function (a, b) {
+            const dateA = new Date(a.joinDate);
+            const dateB = new Date(b.joinDate);
+            return dateA - dateB;
+        });
+        setBeneficiaries(data);
+    };
+
+    const handleClickArchive = () => {
+        setArchivedFilter((prev) => !prev);
+    };
+
+    const handleClickActive = () => {
+        setActiveFilter((prev) => !prev);
     };
 
+    const handleSelectInterest = (selectedValues, selectedItem) => {
+        setNumInterests((prev) => prev + 1);
+        setInterestsFilter(true);
+        switch (selectedItem.id) {
+            case 1:
+                setComputers((prev) => !prev);
+                break;
+            case 2:
+                setSpokenEnglish((prev) => !prev);
+                break;
+            case 3:
+                setLiteracy((prev) => !prev);
+                break;
+            case 4:
+                setMath((prev) => !prev);
+                break;
+            case 5:
+                setBengali((prev) => !prev);
+                break;
+            case 6:
+                setComputers((prev) => !prev);
+                break;
+            case 7:
+                setArts((prev) => !prev);
+                break;
+            case 8:
+                setCounseling((prev) => !prev);
+                break;
+        }
+    };
```

Comment:
Seems a wee bit inefficient - see previous comment about the state for these interests